opal/mca/btl/base/btl_base_frame.c

```diff
@@ -154,6 +156,8 @@ static int mca_btl_base_register(mca_base_register_flag_t flags)
     (void) mca_base_var_enum_register("opal", "btl", "base", "btl_atomic_flags",
                                  &mca_btl_base_atomic_enum);
 
+    (void) mca_base_alias_register ("opal", "btl", "sm", "vader", MCA_BASE_ALIAS_FLAG_NONE);
```

Comment:
This seems like an odd place to put this (i.e., in the base, not in the `sm` component itself).

Is it not sufficient to put this at the very beginning of the `sm` component?

Reply:
That is too late for selection. We don't always dlopen the components before selection so the only place to register the synonym is in the base. My other approach had that issue. Had to make vader static for it to work.

Reply:
Ok.  I added a comment here in the code explaining this.